tests/ada_c.cpp

```diff
@@ -209,3 +209,46 @@ TEST(ada_c, ada_clear_search) {
   ada_free(out);
   SUCCEED();
 }
+
+TEST(ada_c, ada_get_schema_type) {
+  std::string_view input;
+  ada_url out;
+
+  input = "http://www.google.com";
+  out = ada_parse(input.data(), input.size());
+  ASSERT_TRUE(ada_is_valid(out));
+  ASSERT_EQ(ada_get_schema_type(out), 0);
+
+  input = "notspecial://www.google.com";
+  out = ada_parse(input.data(), input.size());
+  ASSERT_TRUE(ada_is_valid(out));
+  ASSERT_EQ(ada_get_schema_type(out), 1);
+
+  input = "https://www.google.com";
+  out = ada_parse(input.data(), input.size());
+  ASSERT_TRUE(ada_is_valid(out));
+  ASSERT_EQ(ada_get_schema_type(out), 2);
+
+  input = "ws://www.google.com/ws";
+  out = ada_parse(input.data(), input.size());
+  ASSERT_TRUE(ada_is_valid(out));
+  ASSERT_EQ(ada_get_schema_type(out), 3);
+
+  input = "ftp://www.google.com/file.txt";
+  out = ada_parse(input.data(), input.size());
+  ASSERT_TRUE(ada_is_valid(out));
+  ASSERT_EQ(ada_get_schema_type(out), 4);
+
+  input = "wss://www.google.com/wss";
+  out = ada_parse(input.data(), input.size());
```

Comment:
Before each parse, can you free the previous allocation?

Reply:
Oh yes, will do! Good point 👍

Reply:
Updated.